src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala

```diff
@@ -468,66 +482,115 @@ private[io] object StageWriter {
-
-    // Save the original COPY command even if additional COPY is run.
-    Utils.setLastCopyLoad(copyStatement.toString)
-
-    // Get missed files if there are any.
-    var missedFileSet = getCopyMissedFiles(params, resultSet, expectedFileSet)
-
-    // If any files are missed, execute 2nd COPY command
-    if (missedFileSet.nonEmpty) {
-      // Negative test:
-      // Only load part of missed files.
-      // Exception is raised for the failure.
-      val secondCopyFileSet: Option[mutable.Set[String]] =
-      if (TestHook.isTestFlagEnabled(
-        TestHookFlag.TH_COPY_INTO_TABLE_MISS_FILES_FAIL)) {
-        Some(missedFileSet.grouped(2).toList.head)
-      } else {
-        Some(missedFileSet)
-      }
-
-      // Generate copy command with missed files only
-      useFilesClause = true
-      val copyWithFileClause = StageWriter.copySql(
-        schema,
-        saveMode,
-        params,
-        targetTable,
-        file,
-        tempStage,
-        format,
-        conn,
-        useFilesClause,
-        secondCopyFileSet.get.toSet
-      )
-
-      def getMissedFileInfo(missedFileSet: mutable.Set[String]): String = {
-        s"""missedFileCount=${missedFileSet.size}
-           | Files: (${missedFileSet.mkString(", ")})
-           |""".stripMargin.filter(_ >= ' ')
+    var lastQueryID: String = null
+    try {
+      // execute the COPY INTO TABLE statement
+      val (resultSet, queryID, errorThrow) =
+        copyStatement.executeWithoutThrow(params.bindVariableEnabled)(conn)
+      lastQueryID = queryID
+      val firstCopyEnd = System.currentTimeMillis()
+      logAndAppend(progress,
+        s"""First COPY command is done in
+           | ${Utils.getTimeString(firstCopyEnd - start)}
+           | at ${LocalDateTime.now()}, queryID is $lastQueryID
+           | ${getResultStatusString(errorThrow)}
+           |""".stripMargin.filter(_ >= ' '))
+      if (errorThrow.isDefined) {
+        throw errorThrow.get
       }
 
-      // Log missed files info
-      log.warn(
-        s"""Some files are not loaded into the table, execute additional COPY
-           | to load them: ${getMissedFileInfo(missedFileSet)}
-           | """.stripMargin)
+      // Save the original COPY command even if additional COPY is run.
+      Utils.setLastCopyLoad(copyStatement.toString)
 
-      // Run the command
-      val resultSet = copyWithFileClause.execute(params.bindVariableEnabled)(conn)
-      missedFileSet = getCopyMissedFiles(params, resultSet, missedFileSet)
+      // Get missed files if there are any.
+      var missedFileSet = getCopyMissedFiles(params, resultSet, expectedFileSet)
 
-      // It is expected all the files must be loaded.
+      // If any files are missed, execute 2nd COPY command
       if (missedFileSet.nonEmpty) {
-        throw new SnowflakeConnectorException(
-          s"""These files are missed when COPY INTO TABLE:
-             | ${getMissedFileInfo(missedFileSet)}
+        // Negative test:
```

Comment:
This seems unrelated.

Reply:
This is not new code. I guess it makes confused because the format change causes it to be like new code.